File: PFE_final/safar_morocco_backend/Safar_Morocco/src/main/java/ma/safar/morocco/offer/service/OfferService.java
package ma.safar.morocco.offer.service;

import ma.safar.morocco.offer.dto.OfferDTO;
import java.util.List;

public interface OfferService {
    OfferDTO createOffer(OfferDTO offerDTO);

    OfferDTO getOfferById(Long id);

    List<OfferDTO> getOffersByDestination(Long destinationId);

    List<OfferDTO> getAllOffers();

    OfferDTO updateOffer(Long id, OfferDTO offerDTO);

    void deleteOffer(Long id);
}

File: PFE_final/frontend/src/app/core/services/offer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Offer } from '../models/offer.model';

@Injectable({
    providedIn: 'root'
})
export class OfferService {
    private apiUrl = '/api/offers';

    constructor(private http: HttpClient) { }

    getOffersByDestination(destinationId: number): Observable<Offer[]> {
        return this.http.get<Offer[]>(`${this.apiUrl}/destination/${destinationId}`);
    }

    getAllOffers(): Observable<Offer[]> {
        return this.http.get<Offer[]>(this.apiUrl);
    }

    createOffer(offer: Offer): Observable<Offer> {
        return this.http.post<Offer>(this.apiUrl, offer);
    }

    updateOffer(id: number, offer: Offer): Observable<Offer> {
        return this.http.put<Offer>(`${this.apiUrl}/${id}`, offer);
    }

    deleteOffer(id: number): Observable<any> {
        return this.http.delete(`${this.apiUrl}/${id}`);
    }
}
